enhancements/machine-api/cluster-api-integration.md

```diff
@@ -0,0 +1,256 @@
+project into OpenShift standalone clusters.
+
+## Motivation
+
+We would like to give users the ablility to use Cluster API for machine management, as an alternative and possible future replacement for Machine API.
+
+### Goals
+
+- Run Cluster API controllers for managing infrastructure in a similar way to Machine API
+- Provide forward compatibility between Machine API (MAPI) and Cluster API (CAPI)
+- Ensure feature parity between MAPI and CAPI before migration
+
+### Non-Goals
+
+- Deprecate or remove any existing APIs
+- Stop providing support for Machine API in near future
+
+## Proposal
+
+In the longer term, our goal is to reconverge with the upstream Cluster API project and leverage the community built providers for managing Machines within OpenShift standalone clusters.
+To gather feedback on the feasability of this goal, we will first introduce Cluster API alongside Machine API as a technical preview, and gather feedback on Cluster APIs feasability as a
+replacement for Machine API.
+
+### User Stories
+
+#### Story 1
+
+As an openshift developer, I would like to leverage the upstream providers and reduce the barrier to OpenShift of supporting new platforms.
+
+#### Story 2
+
+As an openshift developer, I would like to gain the support of third parties who already have vested interests in maintaining Machine providers for various platforms and environments and benefit from their provider specific expertise as I try to add new features.
+
+#### Story 3
+
+As a cloud developer, I would like to easily onboard new platforms as this process is well documented and any implementation will be able to be leveraged by both Kubernetes and OpenShift customers, increasing the value of implementing a new provider.
+
+#### Story 4
+
+As a developer, I would like to be able to use the same set of tools for infrastructure management in OpenShift as I can for vanilla Kubernetes.
+
+### Implementation Details
+
+First, we need to establish Cluster API resource management by ensuring all required components are successfully installed and running within the OpenShift cluster.
+
+Cluster API will only be present in the cluster (installed by a new operator) if and when a user installs a feature gate.
+We will introduce a new, OpenShift specific, feature gate `ClusterAPIEnabled` and include it within the `TechPreviewNoUpgrade` FeatureSet.
+
+Once installed, this preview will allow the user to create new MachineSets using CAPI and explore the features available within CAPI, for comparison with MAPI.
+
+During this timeframe, any user wanting to migrate or try out the preview will be left to manually migrate the MachineSet, using either the upstream documentation or documentation provider by OpenShift.
+
+This preview will be intended to be used to create day 2 worker MachineSets and is not expected to be integrated into the install process in any way.
```

Comment:
This almost sounds like a non-goal to me:

- Provide any automated integration or migration between MAPI and CAPI resources.

Reply:
I will add this as a non-goal, at this stage we expect users to have CAPI resources running together with MAPI.